cmake/Dependencies.cmake

```diff
@@ -539,6 +539,12 @@ if(USE_XNNPACK AND NOT USE_SYSTEM_XNNPACK)
     set(__caffe2_CMAKE_POSITION_INDEPENDENT_CODE_FLAG ${CMAKE_POSITION_INDEPENDENT_CODE})
     set(CMAKE_POSITION_INDEPENDENT_CODE ON)
 
+    if(Win32)
```

Comment:
@xuhancn , do you think it is better to check MSVC rather than Win32? And the same question to [142245](https://github.com/pytorch/pytorch/pull/142245)

https://learn.microsoft.com/en-us/cpp/porting/upgrade-your-code-to-the-universal-crt?view=msvc-170

Reply:
@EikanWang Check `Win32` is fine, it means the target is Windows: https://cmake.org/cmake/help/latest/variable/WIN32.html 
On Windows every compiler need to use Microsoft released SDK.